File: src/main/java/pl/edu/pw/elka/mtoporow/cevolver/lib/model/microstrip/MicrostripLineModelFactory.scala
package pl.edu.pw.elka.mtoporow.cevolver.lib.model.microstrip

import java.util
import java.util.Random

import org.uncommons.watchmaker.framework.CandidateFactory

/**
 * Klasa X
 * Data utworzenia: 06.05.15, 21:03
 * @author Michał Toporowski
 */
@Deprecated
class MicrostripLineModelFactory extends CandidateFactory[MicrostripLineModel] {
  override def generateInitialPopulation(populationSize: Int, rng: Random): util.List[MicrostripLineModel] = ???

  override def generateInitialPopulation(populationSize: Int, seedCandidates: util.Collection[MicrostripLineModel], rng: Random): util.List[MicrostripLineModel] = ???

  override def generateRandomCandidate(rng: Random): MicrostripLineModel = ???
}

File: src/main/java/pl/edu/pw/elka/mtoporow/cevolver/lib/model/microstrip/MicrostripLineModelFactory.java
package pl.edu.pw.elka.mtoporow.cevolver.lib.model.microstrip;

import pl.edu.pw.elka.mtoporow.cevolver.algorithm.datasets.DataHolder;
import pl.edu.pw.elka.mtoporow.cevolver.algorithm.datasets.MeasurementParams;
import pl.edu.pw.elka.mtoporow.cevolver.lib.model.Distances;
import pl.edu.pw.elka.mtoporow.cevolver.lib.model.LWDists;

/**
 * Fabryka modeli linii mikropaskowej
 * Data utworzenia: 04.03.16, 11:13
 *
 * @author Michał Toporowski
 */
public class MicrostripLineModelFactory {

    /**
     * Tworzy nowy model typu kompatybilnego z aktualnym zestawem danych
     *
     * @param distances wektor odległości
     * @return nowy obiekt modelu
     */
    public static MicrostripLineModel newModel(final Distances distances) {
        final MeasurementParams mp = DataHolder.getCurrent().measurementParams();
        final MicrostripParams params = mp.getMicrostripParams();
        switch (mp.getModelType()) {
            case SHORTBREAK:
                return new ShortbreakLineModel(LWDists.shortbreak(distances), params);
            case LONGBREAK:
                return new FixedWidthLineModel(distances, params);
            default:
                throw new IllegalArgumentException("Nieznany model");
        }
    }
}
